jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/AddDataCaseFileInstanceCommand.java

```diff
@@ -105,12 +106,14 @@ public Void execute(Context context) {
                     ProcessInstance pi = (ProcessInstance) ksession.getProcessInstance(processInstanceId);
                     if (pi != null) {
                         ProcessEventSupport processEventSupport = ((InternalProcessRuntime) ((InternalKnowledgeRuntime) ksession).getProcessRuntime()).getProcessEventSupport();
+                        VariableScope variableScope = ((VariableScope) ((RegistryContext) context).get(VariableScope.VARIABLE_SCOPE));
                         for (Entry<String, Object> entry : parameters.entrySet()) {  
                             String name = "caseFile_" + entry.getKey();
                             processEventSupport.fireAfterVariableChanged(
                                 name,
                                 name,
                                 null, entry.getValue(), 
+                                variableScope == null ? null : variableScope.tags(name),
```

Comment:
wouldn't this be safer to add empty list instead of null?

Reply:
Sure, fixed